### ISSUE TITLE
Need to support skipping dns request when  acl  enabled 

### ISSUE THREAD

**Issue by Hjemsokt** [hashicorp/consul#10896]
#### Feature Description
Support dns request and  acl  at the same time  

#### Use Case(s)
 Nessus report : A remote, unauthenticated attacker may able to access Consul Web UI and API to gather data, register services and gain remote access. 
We can add acls to avioid this problem, but we still need dns request at the same time,could it be supported at the same time

**Comment by jkirschner-hashicorp** [comment]
Hi @Hjemsokt,

**Q:** What pages in our documentation did you look for information on whether (and how) this was possible (DNS with ACL)? Perhaps there's an opportunity to improve our documentation in this area.

---

I actually have a [pull request open](https://github.com/hashicorp/consul/pull/10837/files) that attempts to clarify how to use DNS with ACLs. Please take a look and let me know if there's anything that still seems missing or confusing.

In summary, when a DNS request is made to a Consul agent, the agent will use a previously configured token with the request (since there is no means for the initial DNS request to specify a token):
- The Consul agent's [`default` token](https://www.consul.io/docs/agent/options#acl_tokens_default), if configured
- Otherwise, the built-in [`anonymous` token](https://www.consul.io/docs/security/acl/acl-system#builtin-tokens)
